public/app/features/serviceaccounts/ServiceAccountsListPage.tsx

```diff
@@ -4,41 +4,62 @@ import { Icon, LinkButton, useStyles2 } from '@grafana/ui';
 import { css, cx } from '@emotion/css';
 
 import Page from 'app/core/components/Page/Page';
-import { StoreState, ServiceAccountDTO, AccessControlAction } from 'app/types';
-import { loadServiceAccounts, removeServiceAccount, updateServiceAccount } from './state/actions';
+import { StoreState, ServiceAccountDTO, AccessControlAction, Role } from 'app/types';
+import { fetchACOptions, loadServiceAccounts, removeServiceAccount, updateServiceAccount } from './state/actions';
 import { getNavModel } from 'app/core/selectors/navModel';
 import { getServiceAccounts, getServiceAccountsSearchPage, getServiceAccountsSearchQuery } from './state/selectors';
 import PageLoader from 'app/core/components/PageLoader/PageLoader';
-import { GrafanaTheme2 } from '@grafana/data';
+import { GrafanaTheme2, OrgRole } from '@grafana/data';
 import { contextSrv } from 'app/core/core';
+import { UserRolePicker } from 'app/core/components/RolePicker/UserRolePicker';
+import { OrgRolePicker } from '../admin/OrgRolePicker';
 export type Props = ConnectedProps<typeof connector>;
 
-export interface State {}
-
 function mapStateToProps(state: StoreState) {
   return {
     navModel: getNavModel(state.navIndex, 'serviceaccounts'),
     serviceAccounts: getServiceAccounts(state.serviceAccounts),
     searchQuery: getServiceAccountsSearchQuery(state.serviceAccounts),
     searchPage: getServiceAccountsSearchPage(state.serviceAccounts),
     isLoading: state.serviceAccounts.isLoading,
+    roleOptions: state.serviceAccounts.roleOptions,
+    builtInRoles: state.serviceAccounts.builtInRoles,
   };
 }
 
 const mapDispatchToProps = {
   loadServiceAccounts,
+  fetchACOptions,
   updateServiceAccount,
   removeServiceAccount,
 };
 
 const connector = connect(mapStateToProps, mapDispatchToProps);
 
-const ServiceAccountsListPage: React.FC<Props> = ({ loadServiceAccounts, navModel, serviceAccounts, isLoading }) => {
+const ServiceAccountsListPage: React.FC<Props> = ({
```

Comment:
love that we have FC! ❤️ 


it's so hot right now!

Reply:
We are moving away from React.FC in our code and it shouldn't be used for new components. The change is quite simple though:
```
const ServiceAccountsListPage = ({
  loadServiceAccounts,
  fetchACOptions,
  updateServiceAccount,
  navModel,
  serviceAccounts,
  isLoading,
  roleOptions,
  builtInRoles,
}: Props)